compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1294,33 +1274,40 @@ LogicalResult AttentionOp::verify() {
     return op->emitOpError("expected scale to be of floating point type");
   }
 
-  if (failed(checkShapeRank(op, "query", queryType, rankToCompareWith))) {
-    return failure();
-  }
-  if (failed(checkShapeRank(op, "key", keyType, rankToCompareWith))) {
-    return failure();
-  }
-  if (failed(checkShapeRank(op, "value", valueType, rankToCompareWith))) {
-    return failure();
-  }
-  if (failed(checkShapeRank(op, "output", outputType, rankToCompareWith))) {
+  // Check shape compatibility based on indexing maps.
+  SmallVector<int64_t> shape(getIterationDomainRank(), -1);
```

Comment:
The concern is having magic number `-1`; expect it to be different from `ShapedType::kDynamic`. They could have the same value. (Actually `ShapedType::kDynamic` was `-1` in the past, and it is internalized to something else today.) What I'd suggest is having a method to compute the shape first, and then you can compare the result of `affineMap.compose(shape)` and the target tensor shape. It avoids the `-1` magic number.

Reply:
I don't think I have a way of computing the shape without using these indexing maps. I added a `foundDims` array so there is no magic number.